plugin/http/httptrace/httptrace.go

```diff
@@ -59,10 +62,67 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := t.base().RoundTrip(req)
 
 	// TODO(jbd): Add status and attributes.
-	trace.EndSpan(ctx)
+	if err != nil {
+		trace.EndSpan(ctx)
+		return resp, err
+	}
+
+	// trace.EndSpan(ctx) will be invoked after
+	// a read from resp.Body returns io.EOF or when
+	// resp.Body.Close() is invoked.
+	resp.Body = &spanEndBody{rc: resp.Body, spanCtx: ctx}
 	return resp, err
 }
 
+// spanEndBody wraps a response.Body and invokes
+// trace.EndSpan on encountering io.EOF on reading
+// the body of the original response.
+type spanEndBody struct {
+	rc      io.ReadCloser
+	spanCtx context.Context
+
+	endSpanOnce sync.Once
+}
+
+var _ io.ReadCloser = (*spanEndBody)(nil)
+
+func (bpr *spanEndBody) Read(b []byte) (int, error) {
+	n, err := bpr.rc.Read(b)
+	if err == nil {
+		return n, nil
+	}
+
+	// Otherwise, time to end the span or set the status
+	switch err {
+	case io.EOF:
+		bpr.endSpan()
+	default:
+		// For all other errors, set the span status
+		trace.SetSpanStatus(bpr.spanCtx, trace.Status{
+			// Code 2 is for Internal server error as per
+			// https://github.com/googleapis/googleapis/blob/f704d14a7224a140bca5cc26835fae471eaf7281/google/rpc/code.proto#L44-L51
```

Comment:
We can remove the link, source codes are not good source of truths because they often change, removed, reorged, etc.

Reply:
Hmm, but then how do we keep a reference? That link is a permalink btw so won't ever disappear unless it is force pushed or repo deleted.